lib/logstash/outputs/elasticsearch/http_client/manticore_adapter.rb

```diff
@@ -84,19 +84,19 @@ def perform_request(url, method, path, params={}, body=nil)
     end
 
     def format_url(url, path=nil)
-
-      request_uri = url.uri
+      request_uri = url.clone
 
       if path
         # Combine the paths using the minimal # of /s
         # First, we make sure the path is relative so URI.join does
         # the right thing
         relative_path = path && path.start_with?("/") ? path[1..-1] : path
-        # because URI.join obliterates the query parameter, we need to save it
-        # and restore it after this operation
-        query = request_uri.query
-        request_uri = URI.join(request_uri, relative_path)
-        request_uri.query = query if query
+
+        if !request_uri.path
+          request_uri.path = path
+        else
+          request_uri.path = "#{request_uri.path}/#{relative_path}"
```

Comment:
> Do you have a specific concern?

Unknowns.. breaking things that aren't tested and didn't think about upfront. URI, like File paths are tricky things, so it makes sense most transformation of data in a URI to be done by the library instead of outside of it (e.g. Using `File.join(a, b)` instead of `#{a}/#{b}`)